debugger/src/dbm_gdb.c

```diff
@@ -229,34 +229,38 @@ static void gdb_input_write_line(const gchar *line)
 	const char *p;
 	char command[1000];
 	g_snprintf(command, sizeof command, "%s\n", line);
-	
+
 	for (p = command; *p; p += count)
 	{
 		st = g_io_channel_write_chars(gdb_ch_in, p, strlen(p), &count, &err);
 		if (err || (st == G_IO_STATUS_ERROR) || (st == G_IO_STATUS_EOF))
 		{
+			if (err)
+			{
 #ifdef DEBUG_OUTPUT
-			dbg_cbs->send_message(err->message, "red");
+				dbg_cbs->send_message(err->message, "red");
 #endif
-			if (err)
 				g_clear_error(&err);
+			}
 			break;
 		}
 	}
 
 	st = g_io_channel_flush(gdb_ch_in, &err);
 	if (err || (st == G_IO_STATUS_ERROR) || (st == G_IO_STATUS_EOF))
 	{
-#ifdef DEBUG_OUTPUT
-		dbg_cbs->send_message(err->message, "red");
-#endif
 		if (err)
+		{
+			#ifdef DEBUG_OUTPUT
```

Comment:
Actually I was thinking about whether or not but decided  for personal reasons ;)